litho-core/src/main/java/com/facebook/litho/ComponentTree.java

```diff
@@ -2143,7 +2123,29 @@ public LayoutState get() {
         }
       }
 
-      return layoutState.acquireRef();
+      if (released) {
+        if (layoutStateReleased.compareAndSet(false, true)) {
+          layoutState.releaseRef();
+        }
+        return null;
+      } else {
+        return layoutState.acquireRef();
```

Comment:
This is unsafe, because in between checking `released` and calling `acquireRef`, we can release the LayoutState (i.e. someone calls release). I think we're going to need synchronization to get this completely right: I'd still really suggest an approach like I outlined in https://github.com/facebook/litho/pull/417#pullrequestreview-146979425 since it's pretty simple and doesn't involve acquiring a lock for very long :)

Reply:
Ah good catch, so just a lock around the .get() until this acquireRef() to ensure nothing can change between. Will update

Reply:
Let's talk this through on Wednesday in person :)